cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/namespace/DefaultNamespaceAdmin.java

```diff
@@ -169,9 +165,7 @@ public Void call() throws Exception {
     } catch (IOException | ExploreException | SQLException e) {
       // failed to create namespace in underlying storage so delete the namespace meta stored in the store earlier
       nsStore.delete(metadata.getNamespaceId().toId());
-      if (!(Principal.SYSTEM.equals(principal) && NamespaceId.DEFAULT.equals(namespace))) {
```

Comment:
Yup, don't need to check for the SYSTEM principal anymore. 

The user provides: system.user - which defaults to cdap and is granted ADMIN on instance:cdap and ALL on namespace:system. Once this is done, we rely on the fact that the user performing these 'special' operations (creating default namespace, deploying system artifacts, etc) is the user who's been granted privileges during bootstrap. In fact, these aren't 'special' operations anymore. They have the same authorization enforcement.